src/custom/hooks/web3.ts

```diff
@@ -22,30 +23,38 @@ export function useEagerConnect() {
     }
   }, [connector, active])
 
-  const connectInjected = useCallback(() => {
-    // check if the our application is authorized/connected with Metamask
-    injected.isAuthorized().then((isAuthorized) => {
-      if (isAuthorized) {
-        activate(injected, undefined, true).catch(() => {
-          setTried(true)
-        })
-      } else {
-        if (isMobile && window.ethereum) {
+  const connectInjected = useCallback(
+    (providerName = 'Metamask') => {
```

Comment:
It does not exist in `WalletProvider` because it assumes that the injected provider is **Metamask**, I have created an [`enum` ](https://github.com/gnosis/cowswap/blob/2165-adding-wallet-supports/src/custom/hooks/web3.ts#L11) to distinguish between the injected ones.